examples/basicauth/main.go

```diff
@@ -34,14 +35,14 @@ func main() {
 	}
 
 	client := github.NewClient(tp.Client())
-	user, _, err := client.Users.Get("")
+	user, _, err := client.Users.Get(context.Background(), "")
 
 	// Is this a two-factor auth error? If so, prompt for OTP and try again.
 	if _, ok := err.(*github.TwoFactorAuthError); err != nil && ok {
 		fmt.Print("\nGitHub OTP: ")
 		otp, _ := r.ReadString('\n')
 		tp.OTP = strings.TrimSpace(otp)
-		user, _, err = client.Users.Get("")
+		user, _, err = client.Users.Get(context.Background(), "")
```

Comment:
Yeah, using `context.TODO()` can be seen in two ways:

1. It's a TODO for the reader to replace it with a proper context.
2. It's a TODO that we, the library authors, haven't resolved.

I think you want it to be seen as 1, but it might be seen as 2... So I'd rather use `context.Background()` over `context.TODO()` for that reason.